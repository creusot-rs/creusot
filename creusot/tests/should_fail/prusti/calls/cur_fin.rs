extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

#[logic('now)]
#[ensures(result == *x)]
fn cur<'a, X: SnapEq>(x: &'a mut X) -> X {
    *x
}

#[logic('x)]
#[ensures(result == *x)]
fn fin<'now, X: SnapEq>(x: &'now mut X) -> X {
    cur(x)
}
