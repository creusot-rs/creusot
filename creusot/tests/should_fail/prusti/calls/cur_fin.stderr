error[creusot]: expected `&mut _` found `creusot_contracts::prusti::Zombie<&'now mut X>`
  --> cur_fin.rs:13:9
   |
13 |     cur(x)
   |         ^

error: aborting due to previous error

